packages/shared/src/hooks/useAttachments.ts

```diff
@@ -1,47 +1,44 @@
 /* (c) Copyright Frontify Ltd., all rights reserved. */
 
-import { AppBridgeBlock, Asset, useBlockAssets, useBlockSettings } from '@frontify/app-bridge';
+import { AppBridgeBlock, Asset, useBlockAssets } from '@frontify/app-bridge';
 
-export const useAttachments = (appBridge: AppBridgeBlock, blockSettingsKey: string, assetSettingsKey: string) => {
-    const [blockSettings, setBlockSettings] = useBlockSettings(appBridge);
-    const { blockAssets, updateAssetIdsFromKey, addAssetIdsToKey, deleteAssetIdsFromKey } = useBlockAssets(appBridge);
+export const useAttachments = (appBridge: AppBridgeBlock, assetSettingsKey: string) => {
+    const { blockAssets, updateAssetIdsFromKey } = useBlockAssets(appBridge);
     const attachments = blockAssets?.[assetSettingsKey] || [];
-    const attachmentsInBlockSettings = (blockSettings[blockSettingsKey] || []) as { id: number }[];
 
-    const onAddAttachments = async (assets: Asset[]) => {
+    const onAddAttachments = async (newAssets: Asset[]) => {
         const newAssetIds = [];
-        for (const asset of assets) {
+
+        for (const attachment of attachments) {
+            newAssetIds.push(attachment.id);
+        }
+        for (const asset of newAssets) {
             newAssetIds.push(asset.id);
         }
-        await setBlockSettings({
-            [blockSettingsKey]: [...attachmentsInBlockSettings, ...newAssetIds.map((id) => ({ id }))],
-        });
-        await addAssetIdsToKey(assetSettingsKey, newAssetIds);
+        await updateAssetIdsFromKey(assetSettingsKey, newAssetIds);
     };
 
     const onAttachmentDelete = async (assetToDelete: Asset) => {
-        await setBlockSettings({
-            [blockSettingsKey]: attachmentsInBlockSettings.filter(({ id }) => id !== assetToDelete.id),
-        });
-        await deleteAssetIdsFromKey(assetSettingsKey, [assetToDelete.id]);
+        const newAssetIds = [];
+        for (const attachment of attachments) {
+            if (attachment.id !== assetToDelete.id) {
+                newAssetIds.push(attachment.id);
+            }
+        }
+
+        await updateAssetIdsFromKey(assetSettingsKey, newAssetIds);
     };
 
     const onAttachmentReplace = async (attachmentToReplace: Asset, newAsset: Asset) => {
-        const indexOfReplacedAttachment = attachmentsInBlockSettings.findIndex(
-            (attachment) => attachment.id === attachmentToReplace.id
-        );
-        const attachmentsAfterReplace = [...attachmentsInBlockSettings];
-        attachmentsAfterReplace[indexOfReplacedAttachment] = { id: newAsset.id };
-        await setBlockSettings({
-            [blockSettingsKey]: attachmentsAfterReplace,
-        });
         const newAssetIds = [];
+
         for (const attachment of attachments) {
             if (attachment.id !== attachmentToReplace.id) {
                 newAssetIds.push(attachment.id);
+            } else {
+                newAssetIds.push(newAsset.id);
             }
         }
```

Comment:
Could be something like that:

```suggestion
         const newAssetIds = attachments.map((attachment) => {
            if (attachment.id !== attachmentToReplace.id) {
                return attachment.id;
            } else {
                return newAsset.id;
            }
        });
```

Reply:
Good catch. As discussed, for when the assets array comes from the asset chooser it has a custom iterable which does not support map. For the other occasions I updated the code.